acquia/mc-cs-plugin-custom-objects
# MAUT-11467 : fix date issue if empty/not empty filter is used in segment filter

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Yes
| New feature/enhancement? (use the a.x branch)      | No
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | https://acquia.atlassian.net/browse/MAUT-11467 



#### Description:
Segment with custom object filter not working field type is date and operator is empty in filter


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a custom object
3. Add date type field in custom object
4. Create segment
5. Add filter with empty operator on CO field created in step - 2
6. Rebuild segment 
7. Should not throw mysql query error

#### Other areas of Mautic that may be affected by the change:
1. 
2. 

#### List deprecations along with the new alternative:
1. 
2. 



[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1.
2.
